test_conformance/extensions/cl_khr_command_buffer/cl_khr_command_buffer_mutable_dispatch/mutable_command_global_offset.cpp

```diff
@@ -0,0 +1,185 @@
+//
+// Copyright (c) 2022 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include <extensionHelpers.h>
+#include "typeWrappers.h"
+#include "procs.h"
+#include "testHarness.h"
+#include "imageHelpers.h"
+#include <vector>
+#include <iostream>
+#include <random>
+#include <cstring>
+#include <algorithm>
+#include <memory>
+#include "mutable_command_basic.h"
+
+#include <CL/cl.h>
+#include <CL/cl_ext.h>
+
+////////////////////////////////////////////////////////////////////////////////
+// mutable dispatch tests which handle following cases:
+//
+// CL_MUTABLE_DISPATCH_GLOBAL_WORK_OFFSET_KHR
+
+struct MutableDispatchGlobalOffset : InfoMutableCommandBufferTest
```

Comment:
It's a little odd to me that this test is deriving from `InfoMutableCommandBufferTest`.  Why `Info`?

Do we even need `InfoMutableCommandBuffertest`, or would it make more sense to remove it and just get the extension function pointer for `clGetMutableCommandInfoKHR` as part of `BasicMutableCommandBufferTest`?

Reply:
The naming convention was agreed with @EwanC earlier. It's a common part of all of the `mutable dispatch` tests.

Reply:
I had envisaged that just the `mutable_command_info.cpp` tests would inherit from `InfoMutableCommandBufferTest` and the behaviour tests wouldn't be using `clGetMutableCommandInfoKHR` as part of their verification, and so could just inherit from `BasicMutableCommandBufferTest`. But the fact that we do check `clGetMutableCommandInfoKHR` as part of the test verification makes them better tests, so I'm not proposing we change it, and was an oversight on my part.

> Do we even need InfoMutableCommandBuffertest, or would it make more
sense to remove it and just get the extension function pointer for
clGetMutableCommandInfoKHR as part of BasicMutableCommandBufferTest?

I think Ben's suggestion of rolling `clGetMutableCommandInfoKHR` into `BasicMutableCommandBufferTest`
makes sense if we're going to use it in the majority of circumstance, would also make the diff in this PR smaller. Apologies for the confusion, and the overhead of having to make this change in the other PRs